### ISSUE TITLE
Add libraries to javascript kernel

### ISSUE THREAD

**Issue by atolkachiov** [jupyterlite/jupyterlite#439]
### Problem

JupyterLite is awesome, though vanilla Javascript kernel does not allow to do much. 
It would be much more useful to be able to have pre-configured available JS libraries ([lodash](https://lodash.com/), [moment](https://momentjs.com/), etc.) 

### Proposed Solution

Multiple options are available: 
* Provide bootstrap callback, probably in `index.html` config
* Provide the list of urls or local paths of libraries to load in `jupyter-lite.json`

### Additional context

[jupyterlite-p5-kernel](https://github.com/jtpio/jupyterlite-p5-kernel/blob/main/packages/p5-kernel/src/kernel.ts) has custom hardcoded loader for p5.js. It can be implemented in a similar way to avoid creating custom kernels just to preload some libs.

**Comment by bollwyvl** [comment]
Yeah, the proper provisioning of dynamic js, with dependencies is... still hard in 2022. See this long-winded rant here:

https://github.com/jupyterlab/jupyterlab-plugin-playground/issues/1#issuecomment-1008037242

On one of lite's spirit ancestors, [jyve](https://github.com/deathbeds/jyve) i wrapped up a number of existing lab dependencies... but in the post-federated labextension world, this is... more complicated. And then you're pinned to the version lab uses.

As we've learned from the pyodide experience, I think once dependency resolution starts getting involved, an in-browser kernel rapidly needs to become _a distribution_ to not fall apart tomorrow.

For this reason and others, it seems like the vanilla js kernel may well be split off, which started with p5, but will really ramp up when pyodide shelves off on #386. In that repo, or some new/downstream repo in this org, a more full plan of attack could be introduced to handle an opinionated approach for js dependency provisioning.

**Comment by atolkachiov** [comment]
I have doubts about the opinionated approach to dependency management. 

Similar to virtual environments used by server-side kernels, there should be a way to bootstrap a kind of virtual environment for in-browser JS kernel. It does not seem to be reasonable to make JS environment management stuff a part of the kernel distribution. Ability to inject a genetic bootstrap script into JS kernel would be more flexible in my opinion.

**Comment by jtpio** [comment]
We may want to investigate this in the `xeus-javascript` kernel, which has more features than the built-in JavaScript kernel and looks like a more sustainable approach maintenance wise: https://github.com/jupyter-xeus/xeus-javascript

Linking to this issue in particular: https://github.com/jupyter-xeus/xeus-javascript/issues/16
